File: array_problems/2ndLargest2ndSmallestInArray/2ndLargest2ndSmallestInArray.go
package main
import "fmt"
type ArrayProblems struct{}

func (ap *ArrayProblems) arrFunc(array []int) []int {
    for i := 0; i < len(array); i++ {
        for j := 0; j < len(array); j++ {
            if array[i] < array[j] {
                 temp := array[i]
                 array[i] = array[j]
                 array[j] = temp
            }
        }
    }
    min := array[1]
    max := array[len(array) - 2]
    result := []int{ min, max }
	return result
}

func main() {
    arrayProblems := ArrayProblems{}
    secondIndex := arrayProblems.arrFunc([]int { 5, 9, 1, 11, 7, 2, 6, 3, 10, 8, 12, 4 })
    fmt.Println("secondMin:", secondIndex[0])
    fmt.Println("secondMax:", secondIndex[1])
}

File: array_problems/2ndLargest2ndSmallestInArray/2ndLargest2ndSmallestInArray.py
class ArrayProblems:
    def arrFunc(self, arr):
        for i in range(len(arr)):
            for j in range(len(arr)):
                if arr[i] < arr[j]:
                    temp = arr[i]
                    arr[i] = arr[j]
                    arr[j] = temp

        min_index = arr[1]
        max_index = arr[-2]
        result = {
            "secondMin": min_index,
            "secondMax": max_index
        }
        return result


array_problems = ArrayProblems()
secondIndexes = array_problems.arrFunc([5, 9, 1, 11, 7, 2, 6, 3, 10, 8, 12, 4])
print(secondIndexes)